### ISSUE TITLE
Copilot Sidecar Variable by Environment

### ISSUE THREAD

**Issue by hensg** [aws/copilot-cli#3674]
I'm working with copilot trying to enable or disable a sidecar functionality based on the environment in which it is being deployed. So far, I wasn't able to make it work. Any idea? I haven't found anything like this in the documentation either.

Sample:
```yaml
sidecars:
  datadog:
    image: public.ecr.aws/datadog/agent:latest
    variables:
      ECS_FARGATE: true
      DD_SITE: datadoghq.com
      DD_DOGSTATSD_NON_LOCAL_TRAFFIC: true
      DD_APM_ENABLED: true
    secrets:
      DD_API_KEY: XXXXX
    port: 8125
environments:
  staging:
    variables:      
      DD_APM_ENABLED: false
  production:    
    variables:
      DD_APM_ENABLED: true
```

**Comment by Lou1415926** [comment]
Hello @hensg ! The sample that you posted didn't work out probably because it's overriding the `variables` section of the service, not of the `datadog` sidecar. Instead, could you try ⬇️  and see if it works for you?


```yaml
sidecars:
  datadog:
    image: public.ecr.aws/datadog/agent:latest
    variables:
      # ...other vars
      DD_APM_ENABLED: true
environments:
  staging:
    sidecars:
         dotadog:
            variables:      
                  DD_APM_ENABLED: false
```

**Comment by hensg** [comment]
Yep, that worked! Thank you @Lou1415926 , you may close the issue :)

**Comment by Lou1415926** [comment]
Awesome, thanks for letting me know! I'm closing the issue for now!

**Comment by carherr2-cisco** [comment]
> Hello @hensg ! The sample that you posted didn't work out probably because it's overriding the `variables` section of the service, not of the `datadog` sidecar. Instead, could you try ⬇️ and see if it works for you?
> 
> ```yaml
> sidecars:
>   datadog:
>     image: public.ecr.aws/datadog/agent:latest
>     variables:
>       # ...other vars
>       DD_APM_ENABLED: true
> environments:
>   staging:
>     sidecars:
>          dotadog:
>             variables:      
>                   DD_APM_ENABLED: false
> ```

@Lou1415926 , this solved my problem as well. Can you please add this to the documentation? I've parsed both the https://aws.github.io/copilot-cli/docs/developing/sidecars/ and the https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/, and neither make any reference to the ability of being able to pass variables to sidecars based on environments.

**Comment by carherr2-cisco** [comment]
> Hello @hensg ! The sample that you posted didn't work out probably because it's overriding the `variables` section of the service, not of the `datadog` sidecar. Instead, could you try ⬇️ and see if it works for you?
> 
> ```yaml
> sidecars:
>   datadog:
>     image: public.ecr.aws/datadog/agent:latest
>     variables:
>       # ...other vars
>       DD_APM_ENABLED: true
> environments:
>   staging:
>     sidecars:
>          dotadog:
>             variables:      
>                   DD_APM_ENABLED: false
> ```

@Lou1415926 , this solved my problem as well. Can you please add this to the documentation? I've parsed both the https://aws.github.io/copilot-cli/docs/developing/sidecars/ and the https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/, and neither make any reference to the ability of being able to pass variables to sidecars based on environments.

**Comment by carlosaherrera** [comment]
> Hello @hensg ! The sample that you posted didn't work out probably because it's overriding the `variables` section of the service, not of the `datadog` sidecar. Instead, could you try ⬇️ and see if it works for you?
> 
> ```yaml
> sidecars:
>   datadog:
>     image: public.ecr.aws/datadog/agent:latest
>     variables:
>       # ...other vars
>       DD_APM_ENABLED: true
> environments:
>   staging:
>     sidecars:
>          dotadog:
>             variables:      
>                   DD_APM_ENABLED: false
> ```

@Lou1415926 , this solved my problem as well. Can you please add this to the documentation? I've parsed both the https://aws.github.io/copilot-cli/docs/developing/sidecars/ and the https://aws.github.io/copilot-cli/docs/manifest/lb-web-service/, and neither make any reference to the ability of being able to pass variables to sidecars based on environments.

**Comment by Lou1415926** [comment]
@carlosaherrera This is a good idea ❤️  I am reopening this issue with a tag "documentation" so that we know to update the doc with the example!

**Comment by Lou1415926** [comment]
Closed by #5311 !
